From 2492fea2e587fd636435ed20260e588b69449c4b Mon Sep 17 00:00:00 2001
From: Dongfang Zhao <dongfang.zhao@oss.qualcomm.com>
Date: Tue, 28 Oct 2025 14:04:04 +0800
Subject: [PATCH] UPSTREAM: arm64: dts: qcom: Move RPMHCC clocks and
 clock-names to qcs8300.dtsi

Relocate RPMHCC clock and clock-names definitions from board-specific
DTS files to the common qcs8300.dtsi, following the upstream convention.

Fixes a USB data transfer issue on the RB4 base variant when pushing
large files (~600MB+) via ADB. The problem does not occur on custom
variants or the 'qcs8300-ride' machine, indicating a missing RPMHCC
dependency in the base variant DTS.

Signed-off-by: Dongfang Zhao <dongfang.zhao@oss.qualcomm.com>
Upstream-Status: Backport [https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=7be190e4bdd2bd1aca84afef06bb755c06a85473]
---
 arch/arm64/boot/dts/qcom/qcs8300-ride.dts | 5 -----
 arch/arm64/boot/dts/qcom/qcs8300.dtsi     | 4 +++-
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/qcs8300-ride.dts b/arch/arm64/boot/dts/qcom/qcs8300-ride.dts
index 962897610cae..46799169e06c 100644
--- a/arch/arm64/boot/dts/qcom/qcs8300-ride.dts
+++ b/arch/arm64/boot/dts/qcom/qcs8300-ride.dts
@@ -537,11 +537,6 @@ &remoteproc_gpdsp {
 	status = "okay";
 };
 
-&rpmhcc {
-	clocks = <&xo_board_clk>;
-	clock-names = "xo";
-};
-
 &tlmm {
 	dp_hot_plug_det: dp-hot-plug-det-state {
 		pins = "gpio94";
diff --git a/arch/arm64/boot/dts/qcom/qcs8300.dtsi b/arch/arm64/boot/dts/qcom/qcs8300.dtsi
index d619af4603b2..eea77300a194 100644
--- a/arch/arm64/boot/dts/qcom/qcs8300.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs8300.dtsi
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: BSD-3-Clause
 /*
- * Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+ * Copyright (c) 2024-2025 Qualcomm Innovation Center, Inc. All rights reserved.
  */
 
 #include <dt-bindings/clock/qcom,qcs8300-gcc.h>
@@ -4574,6 +4574,8 @@ apps_bcm_voter: bcm-voter {
 			rpmhcc: clock-controller {
 				compatible = "qcom,sa8775p-rpmh-clk";
 				#clock-cells = <1>;
+				clocks = <&xo_board_clk>;
+				clock-names = "xo";
 			};
 
 			rpmhpd: power-controller {
-- 
2.34.1

